### PR TITLE
Add TypeScript to Time Entry Module

### PR DESCRIPTION
**Pull request by naomiaz**

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -14,6 +14,23 @@
     "jsx-a11y/label-has-associated-control": "off",
     "jsx-a11y/anchor-is-valid": "off",
     "arrow-parens": ["error", "always"],
-    "react/destructuring-assignment": "off"
-  }
+    "react/destructuring-assignment": "off",
+    "react/jsx-filename-extension": [1, { "extensions": [".js", ".jsx"] }]
```
Comment (boris-arkenaar) [inline .eslintrc]: What does `1` mean?

File: components/time-entry-form/TimeEntryForm.tsx [hunk 2]
```diff
@@ -1,15 +1,29 @@
...
+import { ClientNamesModel } from '../../ducks/clients';
 import { convertTimeToIso, convertDateToIso, createIsoString } from '../../services/date-time/date-time';
 
 import './time-entry-form.scss';
 
-class TimeEntryForm extends React.Component {
-  static timeEntriesDefaultValues = {
+export interface TimeEntryFormProps {
+  clientNames: ClientNamesModel[];
+  onEntrySubmit;
```
Comment (boris-arkenaar) [inline components/time-entry-form/TimeEntryForm.tsx]: Try defining the type of your functions as well
```
onEntrySubmit: (timeEntry: TimeEntryModel) => void;
```

File: ducks/clients.ts [hunk 3]
```diff
@@ -11,29 +11,60 @@ export const SORT_CLIENTS_DIRECTION = 'SORT_CLIENTS_DIRECTION';
 export const DELETE_CLIENT = 'DELETE_CLIENT';
 export const DELETE_CLIENT_SUCCESS = 'DELETE_CLIENT_SUCCESS';
 
+export interface ClientNamesModel {
```
Comment (boris-arkenaar) [inline ducks/clients.ts]: Use singular, it's only one client name; `ClientNameModel`.

File: ducks/clients.ts [hunk 4]
```diff
@@ -46,14 +77,22 @@ export const getClientsSelector = createSelector(
   )
 );
 
+export const clientNameSelector = createSelector(
```
Comment (boris-arkenaar) [inline ducks/clients.ts]: Here you are actually returning a list of client names, plural; `clientNamesSelector`.

File: ducks/clients.ts [hunk 5]
```diff
@@ -46,14 +77,22 @@ export const getClientsSelector = createSelector(
...
+  clientsSelector,
+  (clients: ClientState['clients']) => clients.reduce((accumulator, currentValue) => ([
```
Comment (boris-arkenaar) [inline ducks/clients.ts]: Use `map` instead of `reduce`.

File: ducks/clients.ts [hunk 6]
```diff
@@ -11,29 +11,60 @@ export const SORT_CLIENTS_DIRECTION = 'SORT_CLIENTS_DIRECTION';
 export const DELETE_CLIENT = 'DELETE_CLIENT';
 export const DELETE_CLIENT_SUCCESS = 'DELETE_CLIENT_SUCCESS';
 
+export interface ClientNamesModel {
+  title: string;
+  value: string;
+}
+
+export interface ClientNames {
```
Comment (boris-arkenaar) [inline ducks/clients.ts]: Are you using this somewhere?

File: components/time-entry-form/TimeEntryForm.tsx (hunk 2 above)
Comment (naomiaz) [reply components/time-entry-form/TimeEntryForm.tsx]: Changed this in all functions

File: ducks/clients.ts (hunk 3 above)
Comment (naomiaz) [reply ducks/clients.ts]: Fixed this

File: ducks/clients.ts (hunk 4 above)
Comment (naomiaz) [reply ducks/clients.ts]: Fixed this

File: ducks/clients.ts (hunk 6 above)
Comment (naomiaz) [reply ducks/clients.ts]: Nope, deleted it therefore

File: ducks/clients.ts (hunk 5 above)
Comment (naomiaz) [reply ducks/clients.ts]: Rewrote this

File: .eslintrc (hunk 1 above)
Comment (naomiaz) [reply .eslintrc]: No idea, but removed the rule anyways as the same rule is described in line 23

**Pull request merged**